client/bio-commandline/src/main/kotlin/uk/ac/ebi/biostd/client/cli/services/SubmissionService.kt

```diff
@@ -1,27 +1,53 @@
 package uk.ac.ebi.biostd.client.cli.services
 
 import ac.uk.ebi.biostd.client.dto.AcceptedSubmission
-import ebi.ac.uk.model.Submission
+import ac.uk.ebi.biostd.client.integration.web.BioWebClient
+import com.github.ajalt.clikt.output.TermUi.echo
+import ebi.ac.uk.coroutines.waitUntil
+import ebi.ac.uk.model.RequestStatus.PROCESSED
 import uk.ac.ebi.biostd.client.cli.dto.DeletionRequest
 import uk.ac.ebi.biostd.client.cli.dto.MigrationRequest
 import uk.ac.ebi.biostd.client.cli.dto.SubmissionRequest
 import uk.ac.ebi.biostd.client.cli.dto.TransferRequest
 import uk.ac.ebi.biostd.client.cli.dto.ValidateFileListRequest
+import java.time.Duration.ofMinutes
+import java.time.Duration.ofSeconds
 
 @Suppress("TooManyFunctions")
 internal class SubmissionService {
-    fun submit(rqt: SubmissionRequest): Submission =
+    suspend fun submit(request: SubmissionRequest): AcceptedSubmission =
         performRequest {
-            val client = bioWebClient(rqt.securityConfig)
-            return client.submitSingle(rqt.submissionFile, rqt.filesConfig).body
+            val client = bioWebClient(request.securityConfig)
+            val acceptedSubmission = client.asyncSubmitSingle(request.submissionFile, request.filesConfig)
+            val (accNo, version) = acceptedSubmission
+
+            echo("SUCCESS: Submission $accNo, version: $version is in queue to be processed")
+
+            if (request.timeout > 0) client.waitForSubmission(request.timeout.toLong(), accNo, version)
+
+            return acceptedSubmission
         }
 
-    fun submitAsync(request: SubmissionRequest): AcceptedSubmission =
-        performRequest {
-            val client = bioWebClient(request.securityConfig)
-            return client.asyncSubmitSingle(request.submissionFile, request.filesConfig)
+    private suspend fun BioWebClient.waitForSubmission(
+        timeout: Long,
+        accNo: String,
+        version: Int,
+    ) {
+        waitUntil(
+            duration = ofMinutes(timeout),
+            interval = ofSeconds(CHECK_INTERVAL),
+        ) {
+            val status = getSubmissionRequestStatus(accNo, version)
+            val isProcessed = status == PROCESSED
+
+            if (isProcessed.not()) echo("INFO: Waiting for submission to be processed")
```

Comment:
I think it will be usefull to show the current status

Reply:
Yeah, I thought the same at first but, since this is something that will be shared with other teams, I don't think it's a good idea to expose our internal logic